be/src/formats/parquet/encoding_dict.h

```diff
@@ -203,10 +204,13 @@ class DictDecoder<Slice> final : public Decoder {
         return Status::OK();
     }
 
-    Status get_dict_values(const std::vector<int32_t>& dict_codes, Column* column) override {
+    Status get_dict_values(const std::vector<int32_t>& dict_codes, const std::vector<uint8_t>& nulls,
+                           Column* column) override {
         std::vector<Slice> slices(dict_codes.size());
         for (size_t i = 0; i < dict_codes.size(); i++) {
-            slices[i] = _dict[dict_codes[i]];
+            if (!nulls[i]) {
```

Comment:
Can we optimize this. I'm afraid it will degrade performance

Reply:
1. if this is column has no null values, it doesn't have to check it every time.
2. split them in batch, check every batch if: 1. all null, 2 all not null, 3 partial null. for the first two case. we don't have to check it every time too,

Reply:
OK. I'll change it.

Reply:
I've updated to check a) all null b) no null 3) partially null.